e2e/nomostest/clean.go

```diff
@@ -126,6 +133,94 @@ func Clean(nt *NT) error {
 	return deleteImplicitNamespacesAndWait(nt)
 }
 
+func uninstallAllPackages(nt *NT) error {
+	// Check if reconciler-manager is reconciled and healthy
+	nt.T.Log("[CLEANUP] checking reconciler-manager status...")
+	rmObj := &unstructured.Unstructured{}
+	rmObj.SetGroupVersionKind(kinds.Deployment())
+	if err := nt.KubeClient.Get(reconcilermanager.ManagerName, configmanagement.ControllerNamespace, rmObj); err != nil {
+		return fmt.Errorf("checking reconciler-manager status: %v", err)
+	}
+	if err := testpredicates.StatusEquals(nt.Scheme, kstatus.CurrentStatus)(rmObj); err != nil {
+		return fmt.Errorf("reconciler-manager not healthy: %v", err)
+	}
+	// If reconciler-manager is healthy, uninstall all the unmanaged RootSyncs & RepoSyncs.
+	// If deletion-propagation is enabled, this will also recursively delete all
+	// managed objects, including other managed packages.
+	if err := uninstallUnmanagedPackagesOfType(nt, kinds.RepoSyncV1Beta1()); err != nil {
+		return fmt.Errorf("uninstalling unmanged RepoSyncs: %v", err)
+	}
+	if err := uninstallUnmanagedPackagesOfType(nt, kinds.RootSyncV1Beta1()); err != nil {
+		return fmt.Errorf("uninstalling unmanged RootSyncs: %v", err)
+	}
+	return nil
+}
+
+func uninstallUnmanagedPackagesOfType(nt *NT, gvk schema.GroupVersionKind) error {
+	// Find all packages
+	nt.T.Logf("[CLEANUP] Listing %s objects...", gvk.Kind)
+	rsObjList := &unstructured.UnstructuredList{}
+	rsObjList.SetGroupVersionKind(kinds.ListGVKForItemGVK(gvk))
+	if err := nt.KubeClient.List(rsObjList, client.InNamespace(configmanagement.ControllerNamespace)); err != nil {
+		return fmt.Errorf("listing %s: %v", gvk.Kind, err)
+	}
+	// Find the unmanaged packages
+	var rsObjs []*unstructured.Unstructured
+	for _, item := range rsObjList.Items {
+		if core.GetAnnotation(&item, metadata.ResourceManagementKey) != metadata.ResourceManagementEnabled {
+			rsObjs = append(rsObjs, &item)
+		} else {
+			nt.T.Logf("[CLEANUP] Skipping deletion of managed %s object %s",
+				gvk.Kind, client.ObjectKeyFromObject(&item))
+		}
+	}
+	// Delete the unmanaged packages in parallel
+	nt.T.Log("[CLEANUP] Deleting unmanaged %s objects...", gvk.Kind)
+	tg := taskgroup.New()
+	for _, obj := range rsObjs {
+		nn := client.ObjectKeyFromObject(obj)
+		if !obj.GetDeletionTimestamp().IsZero() {
+			nt.T.Logf("[CLEANUP] Already terminating %s object %s ...", gvk.Kind, nn)
+		} else {
+			nt.T.Logf("[CLEANUP] Deleting %s object %s ...", gvk.Kind, nn)
+			if err := nt.KubeClient.Delete(obj, client.PropagationPolicy(metav1.DeletePropagationForeground)); err != nil {
+				if apierrors.IsNotFound(err) || meta.IsNoMatchError(err) {
+					// skip waiting
+					nt.T.Logf("[CLEANUP] Confirmed removal of %s object %s", gvk.Kind, nn)
+					continue
+				}
+				// return an error, but don't prevent other deletes or waits
+				tg.Go(func() error {
```

Comment:
I think we need to `continue` after this, otherwise it will add the wait for not found task to the wait group on line 199. If the delete operation failed, I don't think we want to wait for the object to be not found.

Reply:
Added